### DISCUSSION TITLE
Max Size and Performance

### DISCUSSION THREAD

**Discussion by FXIce18** [mlavik1/UnityVolumeRendering#276] [General]
Hello !
Is there a maximum volume size of the data ?
And how does it affect performances ?

The goal is to render a Volume of 150x2740x2740 = 1.13B Voxels, uint16 at 30 fps, including shadows, lighting and material scattering.
Would it be possible ?

Thank you.

FX @ Ice18 Studios

**Comment by mlavik1** [comment]
Hi @FXIce18 !
I believe the maximum resolution of 3D textures is 2048 x 2048 x 2048 (see [here](https://docs.unity3d.com/2021.1/Documentation/Manual/class-Texture3D.html)), so 2740 may be too large. I'd suggest downscaling the dataset in Slicer or something. There's also a setting for that in the Volume Renderer import settings, but it may not look as good.
If the dimension is less than that limit then it should in theory work, as long as you have enough RAM and GPU memory.

**Comment by FXIce18** [comment]
Thank you !
